### PR TITLE
Add `Tree::set_width` and `set_height` functions

### LINKED ISSUE #69: tree.ui sprawls outside of Window when not using CentralPanel.
I'd like to add `egui_tiles` to my app, but it seems Trees don't scale with the ui when they're not in a CentralPanel?

I tested with `examples/simple.rs`, but changed `CentralPanel` to `Window`, and the below is what I got.

```rust

#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")] // hide console window on Windows in release

use egui::Grid;

struct Pane {
    nr: usize,
}

struct TreeBehavior {}

impl egui_tiles::Behavior<Pane> for TreeBehavior {
    fn tab_title_for_pane(&mut self, pane: &Pane) -> egui::WidgetText {
        format!("Pane {}", pane.nr).into()
    }

    fn pane_ui(
        &mut self,
        ui: &mut egui::Ui,
        _tile_id: egui_tiles::TileId,
        pane: &mut Pane,
    ) -> egui_tiles::UiResponse {
        // Give each pane a unique color:
        let color = egui::epaint::Hsva::new(0.103 * pane.nr as f32, 0.5, 0.5, 1.0);
        ui.painter().rect_filled(ui.max_rect(), 0.0, color);

        ui.label(format!("The contents of pane {}.", pane.nr));

        // You can make your pane draggable like so:
        if ui
            .add(egui::Button::new("Drag me!").sense(egui::Sense::drag()))
            .drag_started()
        {
            egui_tiles::UiResponse::DragStarted
        } else {
            egui_tiles::UiResponse::None
        }
    }
}

fn main() -> Result<(), eframe::Error> {
    env_logger::init(); // Log to stderr (if you run with `RUST_LOG=debug`).

    let options = eframe::NativeOptions {
        viewport: egui::ViewportBuilder::default().with_inner_size([320.0, 240.0]),
        ..Default::default()
    };

    let mut tree = create_tree();

    eframe::run_simple_native("My egui App", options, move |ctx, _frame| {
        egui::Window::new("test window")
        //.min_size((250.0, 250.0))
        .show(ctx, |ui| {
            let mut behavior = TreeBehavior {};
            tree.ui(&mut behavior, ui);
        });
    })
}

fn create_tree() -> egui_tiles::Tree<Pane> {
    let mut next_view_nr = 0;
    let mut gen_pane = || {
        let pane = Pane { nr: next_view_nr };
        next_view_nr += 1;
        pane
    };

    let mut tiles = egui_tiles::Tiles::default();

    let mut tabs = vec![];
    tabs.push({
        let children = (0..7).map(|_| tiles.insert_pane(gen_pane())).collect();
        tiles.insert_horizontal_tile(children)
    });
    tabs.push({
        let cells = (0..11).map(|_| tiles.insert_pane(gen_pane())).collect();
        tiles.insert_grid_tile(cells)
    });
    tabs.push(tiles.insert_pane(gen_pane()));

    let root = tiles.insert_tab_tile(tabs);



    egui_tiles::Tree::new("my_tree", root, tiles)
}
```
ui_in_window

### PR DESCRIPTION
**Pull request by rafaga**
I'm trying to make an application where I want the tree don't get all the available space on CentralPanel, Added two new functions to limit the space that takes Tree on the parent container, so you can put widgets outside the tree without using SidePanel or Botton/top Panels.

the proposed functions are 

```
 pub fn set_height(&mut self, height: f32) 

 pub fn set_width(&mut self, width: f32)
``` 

this also resolves #69 because uses `ui.allocate_space(...);` with the resolved rect

### CODE REVIEW

File: src/tree.rs [hunk 1]
```diff
@@ -256,14 +266,37 @@ impl<Pane> Tree<Pane> {
...
+
+    pub fn set_height(&mut self, height: f32) {
+        if height.is_sign_positive() && height.is_finite() {
+            self.height = height;
+        } else {
+            self.height = f32::INFINITY;
+        }
+    }
+
+    pub fn set_width(&mut self, width: f32) {
```
Comment (emilk) [inline src/tree.rs:294]: Please document that the default is to use all available width/height

File: src/tree.rs [hunk 2]
```diff
@@ -37,6 +37,12 @@ pub struct Tree<Pane> {
 
     /// All the tiles in the tree.
     pub tiles: Tiles<Pane>,
+
+    /// Maximum height
+    height: f32,
+
+    /// Maximum width
```
Comment (emilk) [inline src/tree.rs:44]: These docstrings are inaccurate. These are the exact width/height, if finite.

**Review by emilk — CHANGES_REQUESTED** [review]
Remember to add docstrings to public functions

File: src/tree.rs (hunk 2 above)
Comment (rafaga) [reply src/tree.rs]: done

**Review by rafaga — COMMENTED** [review]
done

**Review by emilk — APPROVED** [review]

**Comment by emilk** [comment]
needs a merge fix

File: src/tree.rs [hunk 3]
```diff
@@ -256,18 +266,43 @@ impl<Pane> Tree<Pane> {
...
+    pub fn set_height(&mut self, height: f32) {
+        if height.is_sign_positive() && height.is_finite() {
+            self.height = height;
+        } else {
+            self.height = f32::INFINITY;
+        }
+    }
 
-        // Allocate the used space in the parent Ui:
-        ui.advance_cursor_after_rect(rect);
```
Comment (emilk) [inline src/tree.rs:270]: Bad merge! `allocate_space` should be removed and `advance_cursor_after_rect` re-added

File: src/tree.rs [hunk 4]
```diff
@@ -256,20 +266,45 @@ impl<Pane> Tree<Pane> {
...
 
         self.preview_dragged_tile(behavior, &drop_context, ui);
-
-        // Allocate the used space in the parent Ui:
         ui.advance_cursor_after_rect(rect);
     }
 
+    /// Sets the exact height that can be used by the tree.
+    ///
+    /// determines the height that will be used by the tree component, by default, the tree occupies all the available space in the parent container
```
Comment (emilk) [inline src/tree.rs:288]:
```suggestion
    /// Determines the height that will be used by the tree component.
    /// By default, the tree occupies all the available space in the parent container.
```

File: src/tree.rs [hunk 5]
```diff
@@ -256,20 +266,45 @@ impl<Pane> Tree<Pane> {
...
+        if height.is_sign_positive() && height.is_finite() {
+            self.height = height;
+        } else {
+            self.height = f32::INFINITY;
+        }
+    }
+
+    /// Sets the exact width that can be used by the tree.
+    ///
+    /// determines the width that will be used by the tree component, by default, the tree occupies all the available space in the parent container
```
Comment (emilk) [inline src/tree.rs:299]:
```suggestion
    /// Determines the width that will be used by the tree component.
    /// By default, the tree occupies all the available space in the parent container.
```

**Review by emilk — APPROVED** [review]

**Pull request merged**